airbyte-integrations/connectors/destination-redshift/src/main/kotlin/io/airbyte/integrations/destination/redshift/operation/RedshiftStagingStorageOperation.kt

```diff
@@ -100,7 +100,7 @@ class RedshiftStagingStorageOperation(
             return null
         }
 
-        return generation.first()[JavaBaseConstants.COLUMN_NAME_AB_GENERATION_ID].asLong()
+        return generation.first()[JavaBaseConstants.COLUMN_NAME_AB_GENERATION_ID]?.asLong() ?: 0
```

Comment:
previously we only ever queried the temp raw table for generation ID, so it was always nonnull. Now we have to handle the real raw table (where pre-generationId-migration rows have null generation ID), so add this logic.